### PR TITLE
DMR Fixes and Logging Improvements

### PR DESCRIPTION
**Pull request by guypritchard**
Some fixes/improvements to help with the E2E scenario.

- Logging improvements
- Don't call the classifier to notify of error.
- Some tests

### CODE REVIEW

File: src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs [hunk 1]
```diff
@@ -30,6 +30,8 @@ public Task StartAsync(CancellationToken cancellationToken)
             IsRunning = true;
             StartTimer();
 
+            logger.AsyncProcessorStateChange("AsyncProcessor Started");
```
Comment (decodingahmed) [inline src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs:33]: Just wanted to clarify that the `StartAsync` and `StopAsync` methods (usually) only get called once during the lifetime of the application. It doesn't make sense for these to be logged at "state changed".

File: src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs [hunk 2]
```diff
@@ -67,11 +71,7 @@ private static async void TimerCallback(object? state)
 
             try
             {
-                self.logger.AsyncProcessorStateChange("started");
```
Comment (decodingahmed) [inline src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs:70]: I think these logs should be remain where they are as they capture the start and end of the processing period. Unless you're happy with the logging that happens in the service? Lemme know and I'll close this comment :)

File: src/Dmr.Api/Services/MessageForwarder/MessageForwarderService.cs [hunk 3]
```diff
@@ -56,8 +59,11 @@ public override async Task ProcessRequestAsync(Message payload)
             }
             catch (MessageForwarderException)
             {
-                // If something went wrong - notify the sender.
-                await NotifySenderOfError(payload.Headers).ConfigureAwait(true);
+                // If something went wrong - notify the sender, only if it makes sense to do so.
+                if (payload.Headers.XSentBy != Constants.ClassifierId)
+                {
+                    await NotifySenderOfError(payload.Headers).ConfigureAwait(true);
```
Comment (decodingahmed) [inline src/Dmr.Api/Services/MessageForwarder/MessageForwarderService.cs]:
```suggestion
                    await NotifySenderOfError(payload.Headers).ConfigureAwait(false);
```

**Review by decodingahmed — CHANGES_REQUESTED** [review]
Just requesting some minor changes in the PR. The rest looks good to me 😊

File: src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs (hunk 2 above)
Comment (guypritchard) [reply src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs:70]: Anecdotally, when I was working with these logs they didn't seem to be adding much - they seemed to clutter the logging output with started/completed messages.  Maybe these logs want to be something else?  I don't know...  If you want to keep them we certainly can.

File: src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs (hunk 1 above)
Comment (guypritchard) [reply src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs:33]: Yep - that's what I was seeing.  Just to clarify, Is it the naming you're not keen on or the logging at all?

File: src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs (hunk 2 above)
Comment (decodingahmed) [reply src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs:70]: I'm with you that maybe they are trying to do something else.  My original intention (when I wrote one of the first iterations) for these logs was to track how long it takes for us to process requests. But I suppose we could wrap a stopwatch around this call and do a single log instead.

Side note: would be cool to get the number of processed requests from the `ProcessRequestsAsync` method and log that too. Would give us a clear view of what is going on.

FTR, I think the above thoughts are just thoughts, not suggesting that you do them in this PR

Comment (guypritchard) [reply src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs:70]: Okay - I've added something which will count the number of messages processed and the time taken.  See what you think...

Comment (decodingahmed) [reply src/Dmr.Api/Services/AsyncProcessor/AsyncProcessorHostedService.cs:70]: Lovely - thank you dude!

**Review by decodingahmed — APPROVED** [review]
Le ship-it!

**Review by tonistuulNG — APPROVED** [review]

**Review by decodingahmed — APPROVED** [review]

**Pull request merged**